### PR TITLE
Migrating TrustX adapter to 1.0

### PR DESCRIPTION
**Pull request by PWyrembak**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Other

## Description of change

Migrating TrustX adapter to 1.0

- paul@trustx.org
- [x] official adapter submission

### CODE REVIEW

File: modules/trustxBidAdapter.js [hunk 1]
```diff
@@ -1,165 +1,147 @@
...
+export const spec = {
+  code: BIDDER_CODE,
+  /**
+   * Determines whether or not the given bid request is valid.
+   *
+   * @param {BidRequest} bid The bid params to validate.
+   * @return boolean True if this is a valid bid, and false otherwise.
+   */
+  isBidRequestValid: function(bid) {
+    return !!(bid.params.uid && bid.adUnitCode);
```
Comment (jaiminpanchal27) [inline modules/trustxBidAdapter.js]: `bid.adUnitCode` is not required here to validate bid reqeust. It is part of prebid-core and already tested there.

File: modules/trustxBidAdapter.js [hunk 2]
```diff
@@ -1,165 +1,147 @@
...
+  noAdm: 'Bid from response has no adm parameter - ',
+  noBid: 'Array of bid objects is empty',
+  noPlacementCode: 'Can\'t find in requested bids the bid with auid - ',
+  emptyUids: 'Uids should be not empty',
+  emptySeatbid: 'Seatbid array from response has empty item',
+  emptyResponse: 'Response is empty',
+  hasEmptySeatbidArray: 'Response has empty seatbid array',
+  hasNoArrayOfBids: 'Seatbid from response has no array of bid objects - '
+};
+const bidsRequestMap = {};
```
Comment (jaiminpanchal27) [inline modules/trustxBidAdapter.js]: No global variables. You can create the same in `spec.buildRequests` scope and add it to request object.
`spec.interpretResponse` has request object as second argument.

For reference, check rubiconBidAdapter

File: modules/trustxBidAdapter.js [hunk 3]
```diff
@@ -1,165 +1,147 @@
...
+    bidsRequestMap[bidderRequestId] = bidsMap;
 
-  function _makeHandler(auids, placementMap) {
-    var cbName = bidderCode + '_callback_wrapper_' + auids.join('_');
-    $$PREBID_GLOBAL$$[cbName] = function(resp) {
-      delete $$PREBID_GLOBAL$$[cbName];
-      _responseProcessing(resp, auids, placementMap);
+    const payload = {
+      u: utils.getTopWindowUrl(),
+      pt: window.globalPrebidTrustxPriceType === 'gross' ? 'gross' : 'net',
```
Comment (jaiminpanchal27) [inline modules/trustxBidAdapter.js]: you can pass `window.globalPrebidTrustxPriceType` via param. This is not allowed.

File: modules/trustxBidAdapter.js [hunk 4]
```diff
@@ -1,165 +1,147 @@
...
-        if (bid.params && bid.params.uid) {
-          if (!placementMap[bid.params.uid]) {
-            placementMap[bid.params.uid] = [bid.placementCode];
-            auids.push(bid.params.uid);
-          } else {
-            placementMap[bid.params.uid].push(bid.placementCode);
-          }
-        }
-      }
+    // const payloadString = JSON.stringify(payload);
```
Comment (jaiminpanchal27) [inline modules/trustxBidAdapter.js]: remove unwanted code.

**Comment by PWyrembak** [comment]
@jaiminpanchal27 Thanks, we have updated the code and pulled the changes to the repo. Could you please review once more? Re: bid responses @ test params. Are you calling TX bidder in the browser? I should be definitely seeing bid responses coming from the bidder.

File: modules/trustxBidAdapter.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+Module Type: Bidder Adapter
+Maintainer: paul@trustx.org
+
+# Description
+
+Module that connects to TrustX demand source to fetch bids.
+
+# Test Parameters
+```
+    window.globalPrebidTrustxPriceType = 'gross'; // by default is 'net'
```
Comment (jaiminpanchal27) [inline modules/trustxBidAdapter.md]: After update you don't need this in markdown. Update params as required.

Comment (PWyrembak) [reply modules/trustxBidAdapter.md]: Are you sure you're calling the right prebid location on the page example you've provided? On the page it is "../../build/dev/prebid.js". So I can't see any calls to TX bidder going at all.

**Comment by jaiminpanchal27** [comment]
@PWyrembak 
Apologies for this, but #1742 got merged recently. 

The first argument to interpretResponse now looks like this:
```{
  body: responseBody,
  headers: {
    get: function(header) { /* returns a header from the HTTP response */ }
  }
}
```
You'll have to pull master and update the spec so that it looks digs into that object as well now.

**Pull request merged**